synapse/handlers/typing.py

```diff
@@ -489,16 +489,22 @@ async def get_new_events_as(
             handler = self.get_typing_handler()
 
             events = []
-            for room_id in handler._room_serials.keys():
-                if handler._room_serials[room_id] <= from_key:
+
+            # Work on a copy of things here as these may change when this coroutine awaits the
+            # is_interested_in_room call. Shallow copy is safe as no nested data is present.
```

Comment:
At first I thought this was saying that the call to `service.is_interested_in_room` might change `handler._room_serials`, which sounds like an odd thing for an appservice to do. But I think the point is that Synapse itself might change handler.`_room_serials` while the current coroutine is suspended. Do I understand correctly?

Reply:
Yep that's what I meant! Pushed a better comment in https://github.com/matrix-org/synapse/pull/13392/commits/4fe0b2e31f89b80445b95f3afe451d4abd86374a I think